src/assets/js/dynamic-addon-cards.js

```diff
@@ -130,9 +133,29 @@
             }
           }
 
-          // TODO: we will need to check the min version since the max version
-          // is usually not a problem (AMO does not disable the install button
-          // when we are over the max version).
+          if (!isIncompatible) {
+            const compatibility = current_version.compatibility[clientApp];
```

Comment:
Presumably this data is generally guaranteed and if those expectations aren't met the catch would cover it?

Reply:
`current_version` is checked above but `.compatibility` isn't necessarily checked all the time so I'll add a test case and a condition here.